source/mod_ffield.f90

```diff
@@ -0,0 +1,928 @@
+          write(lout,"(a)") "FFIELD> ERROR Quadrupole element '"//trim(elemName)//"' is not unique."
+          iErr = .true.
+          return
+        end if
+      end do
+      
+      ffQNames(ffNLn) = elemName
+
+      call chr_cast(lnSplit(3),ffQ2File(ffNLn,1),  iErr)
+      call chr_cast(lnSplit(4),ffQ2File(ffNLn,2),  iErr)
+      if(iErr) return
+
+      ffNLn = ffNLn+1
+      
+      if (ffNLn>=ffNLn_max-1) then
+        call ffield_mod_expand_arrays_study(ffNLn_max+10, ffMSn_max, ffNLFile_max)
+      end if
+      
+            
+    case("FFMS") 
+      !     * Name a Multipole that will be skip in the study
+      ! -------------------------------------------------------------------------------------------- !
+      if(nSplit < 2) then
+        write(lout,"(a,i0)") "FFIELD> ERROR FFMS line must have at less 2 values, got ",nSplit
+        iErr = .true.
+        return
+      end if
+
+      elemName = chr_rpad(lnSplit(2), mNameLen)
+
+      ! Check that the name is unique
+      do i=1,ffMSn
+        if(ffMSNames(i) == elemName) then
+          write(lout,"(a)") "FFIELD> ERROR Multipole to skip element '"//trim(elemName)//"' is not unique."
+          iErr = .true.
+          return
+        end if
+      end do
+      
+      ffMSNames(ffMSn) = elemName
+
+      ffMSn = ffMSn+1
+      
+      if (ffMSn>=ffMSn_max-1) then
+        call ffield_mod_expand_arrays_study(ffNLn_max, ffMSn_max+10, ffNLFile_max)
+      end if
+      
+            
+    case("FFFI")   ! FFFI,    File name,     Length equi.,      Total length file
+      !     * Name a File for the study
+      ! -------------------------------------------------------------------------------------------- !
+      if(nSplit < 4) then
+        write(lout,"(a,i0)") "FFIELD> ERROR FFFI line must have at less 4 values, got ",nSplit
+        iErr = .true.
+        return
+      end if
+      
+      elemName = chr_rpad(lnSplit(2), FFFNameLgth)
+      ! Check that the name is unique
+      do i=1,ffNLFile
+        if(ffFNames(i) == elemName) then
+          write(lout,"(a)") "FFIELD> ERROR File '"//trim(elemName)//"' is not unique."
+          iErr = .true.
+          return
+        end if
+      end do
+      
+      ffFNames(ffNLFile) = elemName
+      call chr_cast(lnSplit(3),ffParam(ffNLFile,1),  iErr) ! Lin
+      call chr_cast(lnSplit(4),ffParam(ffNLFile,2),  iErr) ! Lgth
+      if(nSplit>4) call chr_cast(lnSplit(5),ffParam(ffNLFile,3),  iErr) ! Physical aperture (r0)
+      if(iErr) return
+
+      ffNLFile = ffNLFile+1
+      
+      if (ffNLFile>=ffNLFile_max) then
+        call ffield_mod_expand_arrays_study(ffNLn_max, ffMSn_max, ffNLFile_max+10)
+      end if
+      
+    end select
+  
+  end subroutine ffield_parseInputLine
+
+
+
+  ! ================================================================================================ !
+  !  Parse FField Done
+  !  B. Dalena, T. Pugnat and A. Simona from CEA
+  !  V.K. Berglyd Olsen, BE-ABP-HSS
```

Comment:
You still left my name in several places.

Reply:
Done